napari/_vispy/vispy_transforms.py

```diff
@@ -0,0 +1,136 @@
+    transform_chain : vispy.visuals.transforms.ChainTransform
+        Transform chain for an ordered sequence of vispy transforms.
+    translate : ndarray, shape(4)
+        Combined tranlation of all chained transforms, vispy format (X,Y,Z,T).
+    """
+
+    def __init__(self, napari_transform_chain=TransformChain()):
+        self.napari_transform_chain = napari_transform_chain
+
+        self.transform_chain = self._vispy_transform_chain()
+        self._calculate_attributes()
+
+        self.changed = self.transform_chain.changed
+
+        self.napari_transform_chain.events.added.connect(self._add)
+        self.napari_transform_chain.events.removed.connect(self._remove)
+        self.napari_transform_chain.events.reordered.connect(self._reorder)
+
+    @property
+    def scale(self):
+        """Combined scale of all elements in the VispyChainTransform.
+
+        You should not directly modify the scale property of a
+        VispyChainTransform. Instead you should append or insert another
+        transform to the napari TransformChain, which will update this
+        property using event emitters.
+
+        Returns
+        -------
+        ndarray, shape (4,)
+            Combined scale of all elements in the VispyChainTransform.
+            Vispy format expects (X,Y,Z,T)
+        """
+        self._calculate_attributes()
+        return self._scale
+
+    @property
+    def translate(self):
+        """Combined translation of all elements in the VispyChainTransform.
+
+        You should not directly modify the translate property of a
+        VispyChainTransform. Instead you should append or insert another
+        transform to the napari TransformChain, which will update this
+        property using event emitters.
+
+        Returns
+        -------
+        ndarray, shape (4,)
+            Combined translation of all elements in the VispyChainTransform.
+            Vispy format expects (X,Y,Z,T)
+        """
+        self._calculate_attributes()
+        return self._translate
+
+    @property
+    def simplified(self):
+        return self._simplified
+
+    def _vispy_transform_chain(self):
+        """Builds vispy ChainTransform from a napari TransformChain."""
+        vispy_transforms = []
+        for t in self.napari_transform_chain:
+            # From napari format (t, z, y, x)
+            # to the vispy format (x, y, z, ?)
+            _translate = np.flip(t.translate)  # vispy ordered array
+            _scale = np.flip(t.scale)  # vispy ordered array
+            vispy_transforms.append(
+                STTransform(scale=_scale, translate=_translate)
+            )
+        vispy_transform_chain = ChainTransform(vispy_transforms)
+        self.transform_chain = vispy_transform_chain
+        self.__dict__.update(vispy_transform_chain.__dict__)
+        return vispy_transform_chain
+
+    def _calculate_attributes(self):
+        """Simplified transformation matrix, scale, and transle attributes.
```

Comment:
All this attribute stuff is so I can try switching out the STTransform in `_master_transform` for an instance of our new VispyChainTransform class.